### PR TITLE
chore: Informative error for unclosed string

### LINKED ISSUE #273: Non-informative error when updating template with single quotes in attributes
Noticed this error in cloudwatch logs

Can be reproduced with single quotes in the mustached attributes
Screenshot 2020-05-19 at 4 53 09 PM

### PR DESCRIPTION
**Pull request by lshaowei18**
## Problem

Closes #273.

## Solution

This is currently covered in test suite, just that error message isn't informative.

## Before & After Screenshots

**BEFORE**:
82306196-51659000-99f1-11ea-83eb-e6218ee7ebcf

**AFTER**:
Screenshot 2020-05-21 at 9 29 25 AM

### CODE REVIEW

**Comment by lshaowei18** [comment]
Not sure if we should be updating the worker as well, since we are just changing the message.

File: backend/src/core/services/template-client.class.ts [hunk 1]
```diff
@@ -115,6 +115,7 @@ export default class TemplateClient {
     } catch (err) {
       logger.error({ message: `${err.stack}` })
       if (err.message.includes('unclosed tag')) throw new TemplateError('There are unclosed curly brackets in the template')
+      if (err.message.includes('unclosed string')) throw new TemplateError(`Invalid character found in keyword. Only alphanumeric characters allowed.`)
```
Comment (jeantanzj) [inline backend/src/core/services/template-client.class.ts:118]: Maybe something like "Check that your keywords only contain letters and numbers. Keywords like {{ Person's Name }} are not allowed, but {{ PersonsName }} will be allowed."

File: backend/src/core/services/template-client.class.ts [hunk 2]
```diff
@@ -115,6 +115,7 @@ export default class TemplateClient {
     } catch (err) {
       logger.error({ message: `${err.stack}` })
       if (err.message.includes('unclosed tag')) throw new TemplateError('There are unclosed curly brackets in the template')
```
Comment (jeantanzj) [inline backend/src/core/services/template-client.class.ts:117]: Maybe: "Check that all your keywords have double curly brackets around them. A correct example is {{ keyword }}, and incorrect ones are { keyword }, {{ keyword } or { keyword }}.  "

**Comment by lshaowei18** [comment]
Hmmm looks kinda ugly without line breaks
Screenshot 2020-05-21 at 9 13 46 PM

**Comment by jeantanzj** [comment]
Heyyy I think you added the updated message to the wrong error. I suggested changes for both unclosed tag and and unclosed string. 

It’s only a suggestion tho. You can rephrase if you want. I was trying to follow how eslint rules document their rules: eg https://eslint.org/docs/rules/quotes

**Comment by lshaowei18** [comment]
> Heyyy I think you added the updated message to the wrong error. I suggested changes for both unclosed tag and and unclosed string.
> 
> It’s only a suggestion tho. You can rephrase if you want. I was trying to follow how eslint rules document their rules: eg https://eslint.org/docs/rules/quotes

It is pretty good actually, even better if we could have the thumbs up and thumbs down haha

**Pull request merged**